components/eam/src/physics/cam/shoc.F90

```diff
@@ -2855,9 +2855,9 @@ subroutine shoc_assumed_pdf_compute_s(&
   qn=0._rtype
   C=0._rtype
 
-  if (std_s .ne. 0.0_rtype) then
+  if (std_s .gt. 2.0E-155_rtype) then
```

Comment:
You could use `tiny(X)` (with X any number of kind rtype). It seems to be part of the fortran standard, at least f95 and later.

Reply:
Hmm. One question is why std_s would ever approach 0. s is the dry static energy, so is constructed from a sum of temperature and vapor variances... I'm not sure why these would ever be zero. So: are we sure we're getting overflow in realistic cases, or just with random data for unit tests? 

Beyond the question of whether the inputs causing the problem are reasonable, I think your hack is fine as a temporary solution. A better solution might be to notice that the problem calculation always appears as R=s/std_s. The right thing to do would be to use L'Hospital's rule to decide whether s or std_s dominates in this case. I think s should be physically bounded by the temperature and water vapor values seen on planet Earth, while std_s could get arbitrarily large (and is a standard deviation so is always positive), so my bet on L'Hospital is that R will go to infinity as std_s goes to zero. 

Note as well that R is only used in erf(R) and exp(-R) types of expressions, which quickly converge to 1 or 0 as R gets big. This means that we can replace the overflowing R with a big number without impacting the solution. So maybe replace `if (std_s .gt. 2.0E-155_rtype)` with `std_s = min(std_s, tiny(rtype) )`? I think this would give you the C=1 and qn=s  that you're prescribing in the `else` for the `std_s > 2e-155` conditional. So you get rid of the overflow *and* the conditional!

Reply:
@PeterCaldwell , note that `tiny(rtype)` would be somewhere in the range of 1e-308 for doubles, so much much smaller than 2e-155. Since it is the square of `s/std_s` that is overflowing, we'd need to check against something closer to `sqrt(tiny(rtype))`, depending on size of s.

Reply:
Ok, I suggested using tiny(rtype) rather than 2e-155 because hardcoded numbers look ugly rather than for any good reason. The bigger point I was trying to make is that it shouldn't matter whether you use 2e-155 or 1e-30 or tiny(rtype) because all of them will get mapped to approximately identical values of erf(R) or exp(-R). So feel free to replace my ill-advised tiny statements with 2e-155 or whatever value you like.

Reply:
On a different note, shoc.F90 is not going to be supported in scream for much longer, so maybe whatever hack works is fine?

Reply:
But given that you need cxx-f90 bfb-ness, and we want to avoid errors in cxx even when the f90 goes away (if ever), then I encourage a formulation based on a not-hard-coded number. Jim, I think something like `bfb_sqrt(tiny(rtype))` as you suggested is fine.

Reply:
@PeterCaldwell , I was able to come up with a less-hacky min value, `bfb_sqrt(tiny(1._rtype)) * 100`. Trying to get rid of the conditional entirely led to some subtle non-bfbness between f90 and cxx that I'm not sure I want to track down. The cxx version of shoc_assumed_pdf stuff has diverged significantly from fortran.

Reply:
ok, that's fine with me.